### PR TITLE
Added cluster policy selector in the installer

### PR DESCRIPTION
**Pull request by mirhendi**
With this PR, we can specify a cluster policy and create job clusters based on the specified policy

### CODE REVIEW

**Comment by mirhendi** [comment]
fixes issue #656

File: src/databricks/labs/ucx/install.py [hunk 1]
```diff
@@ -419,6 +419,12 @@ def warehouse_type(_):
                     )
                     instance_profile, spark_conf_dict = self._get_ext_hms_conf_from_policy(cluster_policy)
 
+        use_policy = self._question("Do you want to follow a policy to create clusters", default="no")
+        if use_policy != 'no':
+            custom_cluster_policy_id = self._question("Provide a cluster policy id")
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:424]: Can you use _choices()?..

Comment (mirhendi) [reply src/databricks/labs/ucx/install.py]: Added 👍

**Review by nfx — APPROVED** [review]

**Comment by mirhendi** [comment]
Fixed formats and added unit test, please re-execute the workflow.

File: src/databricks/labs/ucx/install.py [hunk 2]
```diff
@@ -482,6 +482,13 @@ def warehouse_type(_):
                     )
                     instance_profile, spark_conf_dict = self._get_ext_hms_conf_from_policy(cluster_policy)
 
+        use_policy = self._question("Do you want to follow a policy to create clusters?", default="no")
```
Comment (dmoore247) [inline src/databricks/labs/ucx/install.py:485]: Too much code in this method, and module. Please extend with a component class. (see `configure.py`)

File: src/databricks/labs/ucx/install.py [hunk 3]
```diff
@@ -971,6 +986,15 @@ def _get_cluster_policies_with_external_hive_metastores(self):
                     yield policy
                     break
 
+    def _check_policy_has_instance_pool(self, policy_id):
```
Comment (dmoore247) [inline src/databricks/labs/ucx/install.py:989]: Too much code in this module, please extend with another module.

Comment (mirhendi) [reply src/databricks/labs/ucx/install.py:989]: Same here, the logic is pretty simple, not sure how to break it

File: src/databricks/labs/ucx/install.py [hunk 4]
```diff
@@ -971,6 +986,15 @@ def _get_cluster_policies_with_external_hive_metastores(self):
...
+        policy = self._ws.cluster_policies.get(policy_id=policy_id)
+        def_json = json.loads(policy.definition)
+        instance_pool = def_json.get("instance_pool_id")
+        if instance_pool is not None:
```
Comment (nfx) [inline src/databricks/labs/ucx/install.py:993]:
```suggestion
        return instance_pool is not None
```
And if-else becomes redundant 😉

**Review by nfx — APPROVED** [review]

**Pull request merged**